parsers/nmparser/parse_lst_file_test.go

```diff
@@ -0,0 +1,70 @@
+package parser
+
+import (
+	"testing"
+)
+
+func TestParseOBJV(t *testing.T) {
+	ofvDetails := new(OfvDetails)
+
+	parseOFV("OBJECTIVE FUNCTION VALUE WITHOUT CONSTANT: 821.705", ofvDetails)
+	if ofvDetails.OFVNoConstant != 821.705 {
+		t.Log("\nGOT: ", ofvDetails.OFVNoConstant, "\n Expected: 821.705")
+		t.Fail()
+	}
+
+	parseOFV("OBJECTIVE FUNCTION VALUE WITH CONSTANT:       1639.561", ofvDetails)
+	if ofvDetails.OFVWithConstant != 1639.561 || ofvDetails.OFVNoConstant != 821.705 {
+		t.Log("\nGOT: ", ofvDetails.OFVWithConstant, "\n Expected: 1639.561")
+		t.Log("\nGOT: ", ofvDetails.OFVNoConstant, "\n Expected: 821.705")
+		t.Fail()
+	}
+
+	parseOFV("N*LOG(2PI) CONSTANT TO OBJECTIVE FUNCTION:    817.855", ofvDetails)
+	if ofvDetails.OFV != 817.855 || ofvDetails.OFVWithConstant != 1639.561 || ofvDetails.OFVNoConstant != 821.705 {
+		t.Log("\nGOT: ", ofvDetails.OFV, "\n Expected: 817.855")
+		t.Log("\nGOT: ", ofvDetails.OFVWithConstant, "\n Expected: 1639.561")
+		t.Log("\nGOT: ", ofvDetails.OFVNoConstant, "\n Expected: 821.705")
+		t.Fail()
+	}
+
+	parseOFV("#OBJV:********************************************    -7913.528       **************************************************", ofvDetails)
+	if ofvDetails.OFV != 817.855 || ofvDetails.OFVWithConstant != 1639.561 || ofvDetails.OFVNoConstant != -7913.528 {
+		t.Log("\nGOT: ", ofvDetails.OFV, "\n Expected: 817.855")
+		t.Log("\nGOT: ", ofvDetails.OFVWithConstant, "\n Expected: 1639.561")
+		t.Log("\nGOT: ", ofvDetails.OFVNoConstant, "\n Expected: 821.705")
+		t.Fail()
+	}
+}
+
+func TestParTestParseOBJV2(t *testing.T) {
+	var lines = []string{
+		"OBJECTIVE FUNCTION VALUE WITHOUT CONSTANT: 821.705",
+		"OBJECTIVE FUNCTION VALUE WITH CONSTANT:       1639.561",
+		"N*LOG(2PI) CONSTANT TO OBJECTIVE FUNCTION:    817.855",
+		"#OBJV:********************************************    -7913.528       **************************************************",
+	}
+	lstData := ParseLstEstimationFile(lines)
+	if lstData.OFV.OFV != 817.855 || lstData.OFV.OFVWithConstant != 1639.561 || lstData.OFV.OFVNoConstant != -7913.528 {
```

Comment:
I'm not a fan of this `||` based testing structure. Namely, it does not scale well, especially if you have more complex structures to compare. In addition, it is unclear if you are testing the entire struct or just _parts_ of a struct. With testify you can just use `assert.Equal()` to compare against initialized structs with the correct value. 

```
expected := OfvDetails{
// correct values
}
assert.Equal(t, lstData.OFV, expected)
```

Reply:
Yeah, assert is way better.